### PR TITLE
Fix on prop bleeding to dom in ToggleSwitch

### PR DESCRIPTION
**Pull request by mdodgelooker**
`<div on={true}` causes a React warning. This solution isn't elegant but it works and I couldn't find any better inspiration in https://github.com/styled-components/styled-components/issues/135

The other props like `size` and `disabled` weren't causing breakage but I filtered those as well since they have no place on a `<div/>`.

### CODE REVIEW

File: packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ export interface ToggleSwitchProps
   size?: number
 }
 
-const Knob = styled.div<KnobProps>`
+const Knob = styled(({ className }) => <div className={className} />)`
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx]: is this actually ever receiving the className? looks like on line 86 the className prop is plucked off the props and applied to `KnobContainerBase` instead

Comment (mdodgelooker) [reply packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx]: That was my initial thought too, but if I don't put that there, it breaks the styles – I guess b/c `styled` creates a new `className`.

**Comment by dbchristopher** [comment]
Wait, disregard that error @mdodgelooker. Just restarted the server dev server and the error went awa

File: packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx [hunk 2]
```diff
@@ -83,26 +83,30 @@ const Knob = styled.div<KnobProps>`
...
 
-const KnobContainerBase = styled.div<KnobProps & PseudoProps>`
+const KnobContainerBase = styled(
+  ({
+    children,
+    className,
+  }: KnobProps & PseudoProps & { children: ReactNode }) => (
+    <div className={className}>{children}</div>
+  )
+)`
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx]: I personally feel like it's cleaner to define a new TS interface rather than inlining the types. E.G.

```
interface KnobContainerBaseProps extends KnobProps, PseudoProps {
  children: ReactNode
}

const KnobContainerBase = styled(({ children, className }) => (
  <div className={className}>{children}</div>
))<KnobContainerBaseProps>`
  ...
`
```

File: packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx (hunk 1 above)
Comment (dbchristopher) [reply packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx]: ah interesting! yeah that makes sense

File: packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx [hunk 3]
```diff
@@ -83,26 +87,25 @@ const Knob = styled.div<KnobProps>`
...
-        size={props.size}
-        on={props.on}
-        ref={ref}
-      >
-        <Knob {...props} />
-      </KnobContainerBase>
-    )
-  }
-)
+const KnobContainer: FC<KnobProps> = ({ className, ...props }) => {
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx]: I played with it a little more locally. I think you can revert the changes to `Knob` if you destructure the `on` prop here too. That way it won't be passed down into `<Knob {...props} />` in the first place

Comment (dbchristopher) [reply packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx]: ah, never mind! then that breaks the functionality altogether. Ok PR approved 😅
